sync-core/src/test/java/com/cloudant/sync/query/IndexTest.java

```diff
@@ -0,0 +1,135 @@
+//  Copyright (c) 2015 Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IndexTest {
+
+    private List<Object> fieldNames;
+    private String indexName;
+
+    @Before
+    public void setUp() {
+        fieldNames = Arrays.<Object>asList("name", "age");
```

Comment:
The list of fieldNames can be a String list or it can be a list that contains Map objects that provide the filename as well as an index sort direction.  For example: a Map object could be `{ "fieldname" : "DESC" }` or `{ "filename" : "ASC" }`.